pennylane/gradients/parameter_shift.py

```diff
@@ -256,6 +257,7 @@ def expval_param_shift(tape, argnum=None, shifts=None, gradient_recipes=None, f0
         gradient_data.append((len(g_tapes), coeffs, None, unshifted_coeff))
 
     def processing_fn(results):
+
         # Apply the same squeezing as in qml.QNode to make the transform output consistent.
         # pylint: disable=protected-access
         if tape._qfunc_output is not None and not isinstance(tape._qfunc_output, Sequence):

```

Comment:
Here it covers the case for single measurement and ndarray? The second part is new from PR3, we might have to change the cond here.